pkg/mls/api/v1/service.go

```diff
@@ -371,6 +388,43 @@ func (s *Service) SubscribeGroupMessages(req *mlsv1.SubscribeGroupMessagesReques
 		defer func() {
 			_ = sub.Unsubscribe()
 		}()
+
+		if filter.IdCursor > 0 {
```

Comment:
Hopefully I'm understanding this correctly - this is happening sequentially after the streaming from the relay is set up, so some of the newer messages from the relay might come in before we deliver these older messages?

I guess the reason for this ordering is to avoid gaps? And it might be possible to receive the messages from the relay, but defer pushing them until after the sync from DB is complete, but it's unnecessarily complex?

Reply:
Yep exactly. This is just the simplest approach, but we could defer the relayed messages with the tradeoff of complexity and resource/memory usage.

Reply:
Because Relay itself is loosely ordered it wouldn't change the overall guarantees. This particular case just leads to messages that might be _really_ out of order, instead of maybe off by one or two places.